modules/SystemTest/src/server/routes/SystemTestRoutes.coffee

```diff
@@ -86,44 +147,102 @@ exports.getOrCreateACASBobInternal = (callback) ->
 				callback statusCode, {
 					hasError: hasError
 					messages: body
-					created: created
 				}
 
-exports.deleteACASBob = (req, resp) ->
-	exports.deleteACASBobInternal (statusCode, output) ->
-		resp.statusCode = statusCode
-		resp.json output
-
-exports.deleteACASBobInternal = (callback) ->
-	exports.getACASBobUser (bob) ->
-		if !bob?
-			callback 400, {
-				hasError: true
-				messages: bob
-			}
-		else
-			options =
-				method: 'DELETE'
-				url: "#{config.all.client.service.persistence.fullpath}authors/#{bob.id}"
-				json: true
-			request options, (error, response, body) ->
+add_project_roles = (projectNames, roles, username, callback) ->
+	if !projectNames?
+		projectNames = []
+	# Look up projects and translate names to codes
+	options =
+		method: 'GET'
+		json: true
+		url: "#{config.all.server.nodeapi.path}/api/projects/getAllProjects/stubs"
+	request options, (error, response, body) ->
+		if error
+			throw new Error(error)
+		_.each projectNames, (projectName) ->
+			project = _.findWhere body, {"name":projectName}
+			if project?
+				roles.push {
+					roleType: 'Project'
+					roleKind: project.code
+					roleName: 'User'
+					userName: username
+				}
+		callback roles
+
+# Grant roles to a test user
+exports.giveTestUserRolesInternal = (username, acas_user, acas_admin, cmpdreg_user, cmpdreg_admin, projects, callback) ->
+	roles = []
+	if acas_user
+		roles.push {
+			roleType: 'System'
+			roleKind: 'ACAS'
+			roleName: 'ROLE_ACAS-USERS'
+			userName: username
+		}
+	if acas_admin
+		roles.push {
+			roleType: 'System'
+			roleKind: 'ACAS'
+			roleName: 'ROLE_ACAS-ADMINS'
+			userName: username
+		}
+	if cmpdreg_user
+		roles.push {
+			roleType: 'System'
+			roleKind: 'CmpdReg'
+			roleName: 'ROLE_CMPDREG-USERS'
+			userName: username
+		}
+	if cmpdreg_admin
+		roles.push {
+			roleType: 'System'
+			roleKind: 'CmpdReg'
+			roleName: 'ROLE_CMPDREG-ADMINS'
+			userName: username
+		}
+	add_project_roles projects, roles, username, (updated_roles) ->
+		options =
+			method: 'POST'
+			url: "#{config.all.client.service.persistence.fullpath}authorroles/saveRoles"
+			json: true
+			body: roles
+		request options, (error, response, body) ->
 				if error
-					console.error response
+					throw new Error(error)
 					statusCode=500
 					hasError = true
 				else
 					if response.statusCode == 500
 						hasError = true
-						created = false
 					else
 						hasError = false
-						created = true
 					statusCode = response.statusCode
 				callback statusCode, {
 					hasError: hasError
 					messages: body
 				}
 
+# Legacy "ACAS Bob" routes
```

Comment:
Made all the "bob" routes simple passthroughs to the more flexible new functions, just with hardcoded values for "bob"